packages/client/src/devices/MicrophoneManager.ts

```diff
@@ -58,21 +63,37 @@ export class MicrophoneManager extends InputMediaDeviceManager<MicrophoneManager
   }
 
   private async startSpeakingWhileMutedDetection(deviceId?: string) {
+    await this.stopSpeakingWhileMutedDetection();
+
     if (isReactNative()) {
-      return;
+      this.rnSpeechDetector = new RNSpeechDetector();
+      await this.rnSpeechDetector.start();
+      const unsubscribe = this.rnSpeechDetector?.onSpeakingDetectedStateChange(
+        (event) => {
+          this.state.setSpeakingWhileMuted(event.isSoundDetected);
+        },
+      );
+      this.soundDetectorCleanup = () => {
+        unsubscribe();
+        this.rnSpeechDetector?.stop();
+        this.rnSpeechDetector = undefined;
+      };
+    } else {
+      // Need to start a new stream that's not connected to publisher
+      const stream = await this.getStream({
+        deviceId,
+      });
+      this.soundDetectorCleanup = createSoundDetector(stream, (event) => {
+        this.state.setSpeakingWhileMuted(event.isSoundDetected);
+      });
     }
-    await this.stopSpeakingWhileMutedDetection();
-    // Need to start a new stream that's not connected to publisher
-    const stream = await this.getStream({
-      deviceId,
-    });
-    this.soundDetectorCleanup = createSoundDetector(stream, (event) => {
-      this.state.setSpeakingWhileMuted(event.isSoundDetected);
-    });
   }
 
   private async stopSpeakingWhileMutedDetection() {
-    if (isReactNative() || !this.soundDetectorCleanup) {
+    if (isReactNative()) {
+      this.rnSpeechDetector?.stop();
+    }
```

Comment:
This block is redundant too. `this.soundDetectorCleanup` will stop the RN detector.

Reply:
Done